---
title: "R Notebook"
output: html_notebook
---


Step 1: data input and construct a DESeqDataSet
```{r}
library("tximport")
library("readr")
dir <- system.file("extdata", package="tximportData")
counts <- read.table(file.path(dir,"sample003.csv"), header=TRUE)
counts$condition <- factor(rep(c("TNB","HR"),each=20))
rownames(counts) <- samples$run
counts[,c("TNB_sample1","TNB_sample2","TNB_sample3","TNB_sample4")]
```

```{r}
sample003 <- read_csv("/users/batulal-zubeidy/Desktop/TRGNFinal/TNB_HR_Joined_counts.csv")
```

```{r}
head(sample003)
```

```{r}
counts <- read.csv("/Users/batulal-zubeidy/Desktop/TRGNFinal/TNB_HR_Joined_Counts.csv", header = TRUE, row.names = 1, sep = ",")
```

```{r}
head(counts)
```


```{r}
library("DESeq2")
dds <- DESeqDataSetFromTximport(txi,
                                   colData = samples,
                                   design = ~ condition)
```

```{r}
samplesFiles <- grep("TNB_sample1",list.files("TNB_HR_Joined"), value = TRUE)

```

```{r}
counts <- counts[which(rowSums(counts) >1),]
```

```{r}
condition <- factor(c("TNB_Sample1", "TNB_Sample2", "TNB_Sample3", "TNB_Sample4", "TNB_Sample5", "TNB_Sample6", "TNB_Sample7", "TNB_Sample8", "TNB_Sample9", "TNB_Sample10", "HR_Sample1", "HR_Sample2", "HR_Sample3", "HR_Sample4", "HR_Sample5", "HR_Sample6", "HR_Sample7", "HR_Sample8", "HR_Sample9", "HR_Sample10"))
```

```{r}
coldata
```

```{r}
rownames(coldata) <- sub("fb", "", rownames(coldata))
all(rownames(coldata) %in% colnames(cts))
```

```{r}
all(rownames(coldata) == colnames(cts))
```

```{r}
cts <- cts[, rownames(coldata)]
all(rownames(coldata) == colnames(cts))
```

```{r}
dds <- DESeqDataSetFromMatrix(countData = round(counts), colData = metadata, design = ~condition)
```

```{r}
library("DESeq2")
dds <- DESeqDataSetFromMatrix(countData = round(counts), colData = metadata, design = ~condition)
```

```{r}
dds <- DESeq(dds)
```

```{r}
featureData <- data.frame(gene=rownames(counts))
mcols(dds) <- DataFrame(mcols(dds), featureData)
mcols(dds)
```

Step 2: gene differential expression analysis

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

# Fitting data into the model for analysis
```{r}
res <- results(dds, contrast = c("condition", "TNB", "HR"))
```

```{r}
res
```

```{r}
library("BiocParallel")
register(MulticoreParam(4))
```


p-values and adjusted p-values
```{r}
resOrdered <- res[order(res$pvalue),]
```

```{r}
summary(res)
```

```{r}
sum(res$padj < 0.1, na.rm=TRUE)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

```{r}
sum(res05$padj < 0.05, na.rm=TRUE)
```

Independent hypothesis weighting: 
```{r}
# (unevaluated code chunk)
library("IHW")
resIHW <- results(dds, filterFun=ihw)
summary(resIHW)
sum(resIHW$padj < 0.1, na.rm=TRUE)
metadata(resIHW)$ihwResult
```

Exploring and exporting results: 
MA-plot

```{r}
plotMA(res, ylim=c(-2,2))
```

The lower left or upper right of the dots represent the more abundant and variable genes

```{r}
plotMA(resLFC, ylim=c(-5,5))
```

Alternative shrinkage estimators:

```{r}
BiocManager::install("apeglm")
```

```{r}
install.packages("CRAN")
```

```{r}
resultsNames(dds)
```


```{r}
resLFC <-lfcShrink(dds, coef=2, type="apeglm")
```


```{r}
resLFC
```

```{r}
install.packages("ashr")
```


```{r}
resAsh <- lfcShrink(dds, coef=2, type="ashr")
```


```{r}
# because we are interested in treated vs untreated, we set 'coef=2'
resNorm <- lfcShrink(dds, coef=2, type="normal")
```


```{r}
par(mfrow=c(1,3), mar=c(4,4,2,1))
xlim <- c(1,1e5); ylim <- c(-3,3)
```


```{r}
plotMA(resLFC, xlim=xlim, ylim=ylim, main="apeglm")
plotMA(resNorm, xlim=xlim, ylim=ylim, main="normal")
plotMA(resAsh, xlim=xlim, ylim=ylim, main="ashr")
```


```{r}
#idx <- identify(res$baseMean, res$log2FoldChange)
#rownames(res)[idx]
```

Plot counts

```{r}
plotCounts(dds, gene=which.min(res$padj), intgroup="condition")
```

```{r}
d <- plotCounts(dds, gene=which.min(res$padj), intgroup="condition", 
                returnData=TRUE)
library("ggplot2")
ggplot(d, aes(x=condition, y=count)) + 
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  scale_y_log10(breaks=c(25,100,400))
```

```{r}
mcols(res)$description
```

Exporting results to CSV files


```{r}
resSig <- subset(resOrdered, padj < 0.1)
resSig
```

```{r}
write.csv(as.data.frame(resOrdered), 
          file="TNB_vs_HR_results.csv")
```

Data transformations and visualization:

```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```

```{r}
#extract the matrix of normalized values.
```

```{r}
BiocManager::install("vsn")
```

Effects of transformations on the variance
```{r}
# this gives log2(n + 1)
ntd <- normTransform(dds)
library("vsn")
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
meanSdPlot(assay(rld))
```

Heatmap of the count matrix:

```{r}
mat.z <- t(apply(mat, 1, scale))
colnames(mat.z) <- rownames(coldata)
##Testing code below
testing <- mat.z
testing <- as.data.frame(testing)
testing$ensembl <- row.names(testing)
library(dplyr)

```

```{r}
joined <- rename (Joined, 'ensembl' = 'V1')
testing2 <- full_join(joined, testing, by = "ensembl")
testing2 <- testing2[-c(1)]
row.names(testing2) <- testing2$V2
testing2 <- testing2[-c(1)]
testing2 <- as.matrix(testing2)
```

```{r}
install.packages("stringr")
```


```{r}
library("pheatmap")
library("stringr")
select <- order(rowMeans(counts(dds,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df <- as.data.frame(colData(dds)[,c("condition")])
colnames(assay(vsd)) <- str_sub(colnames(assay(vsd)), 1, 21)
rownames(df) <- colnames(assay(vsd))
pheatmap(assay(vsd)[select,],cellwidth = 4,cellheight = 8,fontsize_row = 9,fontsize_col =4, annotation_col=df)

```

These are the highest 20 genes expressed in the TNB and HR-positive breast cancer patients.  The dark oranage to red represents the highest expression, while the dark blue is the lowerest expression.  


```{r}
pheatmap(assay(rld)[select,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=FALSE, annotation_col=df)
```

Heatmap of the sample-to-sample distances
```{r}
sampleDists <- dist(t(assay(vsd)))
```

```{r}
library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition, vsd$type, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

This heatmap gives an idea of how similar are the samples in terms of gene expression. 


Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Cmd+Option+I*.

Principal component plot of the samples

```{r}
plotPCA(vsd, intgroup=c("condition"))
```

Conclusion:
1. Triple-negative breast cancer in females appeared to show significant differences in gene expression relative to hormone-receptor-positive breast cancers. ![heatmap] (https://github.com/alzubeid/TRGN-510-FINAL-PROJECT/blob/main/Image%2012-4-22%20at%209.34%20PM.jpeg)

2. The principal component separation plot showed that the two cohorts significantly different. 

3. These differences may overestimate the true genetic alterations between the histological tissue. This overestimation may be due to racial, age-related, or hormonal status.

4. Further study including larger sample sizes should be performed, where race, hormonal status, and stage can be compared.

Known Issues:
-Small sample sizes were used to run the test for differential expression, which likely obsecures the true gene expression.
-The data obtained from TCGA cases were cross examined with portal.gdc.cancer.gov dataset to obtain the clinical pathology, including estrogn/progestrone/HER2 biomarkers.
-The RNA-sequence data showed significantly different PCA analyses between triple negative and estorgen-receptor-positive breast cancer; much greater differences than anticipated.
